docs/DeveloperGuide.adoc

```diff
@@ -385,7 +385,55 @@ We are using `java.util.logging` package for logging. The `LogsCenter` class is
 
-=== [Proposed]
+=== [Proposed] Calendar Viewer Feature
+==== Current Implementation
+The Calendar Viewer mechanism is facilitated by `Google Calendar API` and reside in the `ModelManager`.
+It supports viewing/add/editing/deleting capability that modifies the state of the NUSCouples App.
+Firstly, it uses Outh 2.0 endpoints to allow users to share specific data with the application while keeping their usernames, passwords, and other information private. For example, an application can use OAuth 2.0 to obtain permission from users to store files in their Google Drives which sync to the calendar.
+The only downside to this implementation is that the user needs to connect to the internet because NUSCouples app needs to open the system browser and supply a local redirect URI to handle responses from Google's authorization server.
+
+*Basic steps*
+
+All applications follow a basic pattern when accessing a Google API using OAuth 2.0. At a high level, this are the four steps:
+
+1. Obtain OAuth 2.0 credentials from the Google API Console.
+Visit the Google API Console to obtain OAuth 2.0 credentials such as a client ID and client secret that are known to both Google and your application. The set of values varies based on what type of application you are building. For example, a JavaScript application does not require a secret, but a web server application does.
+
+2. Obtain an access token from the Google Authorization Server.
+Before your application can access private data using a Google API, it must obtain an access token that grants access to that API. A single access token can grant varying degrees of access to multiple APIs. A variable parameter called scope controls the set of resources and operations that an access token permits. During the access-token request, your application sends one or more values in the scope parameter.
+There are several ways to make this request, and they vary based on the type of application you are building. For example, a JavaScript application might request an access token using a browser redirect to Google, while an application installed on a device that has no browser uses web service requests.
+Some requests require an authentication step where the user logs in with their Google account. After logging in, the user is asked whether they are willing to grant the permissions that your application is requesting. This process is called user consent.
+If the user grants the permission, the Google Authorization Server sends your application an access token (or an authorization code that your application can use to obtain an access token). If the user does not grant the permission, the server returns an error.
+It is generally a best practice to request scopes incrementally, at the time access is required, rather than up front. For example, an app that wants to support purchases should not request Google Wallet access until the user presses the “buy” button; see Incremental authorization.
+
```

Comment:
Too wordy? break up into bullet points using * or use diagram to explain to make it easier to understand. can label steps 1 - 3 in the diagram so that reader understands better

Reply:
yes mdm